sql/hive/v0.12.0/src/main/scala/org/apache/spark/sql/hive/Shim12.scala

```diff
@@ -135,7 +135,13 @@ private[hive] object HiveShim {
       PrimitiveCategory.VOID, null)
 
   def getStringWritable(value: Any): hadoopIo.Text =
-    if (value == null) null else new hadoopIo.Text(value.asInstanceOf[UTF8String].toString)
+    if (value == null) {
+      null
+    } else if (value.isInstanceOf[String]) {
+      new hadoopIo.Text(value.asInstanceOf[String])
+    } else {
+      new hadoopIo.Text(value.asInstanceOf[UTF8String].toString)
+    }
```

Comment:
That's actually my concern, not every developer knows exactly how the `foldable` work in Hive UDF, I don't think we need to implement this Hive UDF at all. We can just return the `Literal(database, StringType)` in `HiveFunctionRegistry.lookupFunction` instead of creating the Hive UDF and register it, which seems a hack way to me.